proposed-html-structure-from-wordpress.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+class Research_Guide {
+	private $taxonomies;
+	private $categories;
+	private $tags;
+	private $guidance;
+	private $url;
+	private $title;
+	private $slug;
+	private $id;
+	private $recommended;
+	private $partners;
+	private $online;
+
+	public function __construct($post) {
+		$this->set_url($post);
+		$this->set_title($post);
+		$this->set_slug($post);
+		$this->set_id($post);
+		$this->set_taxonomies($post);
+		$this->set_categories($post);
+		$this->set_tags($post);
+		$this->set_guidance($post);
+		$this->set_recommended();
+		$this->set_online();
+		$this->set_partners();
+	}
+
+	function set_url ($post) {
+		$this->url = get_permalink($post);
+	}
+	function set_title ($post) {
+		$this->title = get_the_title($post);
+	}
+	function set_slug ($post) {
+		$this->slug = $post->post_name;
+	}
+	function set_id ($post) {
+		$this->id = $post->ID;
+	}
+	function set_taxonomies ($post) {
+		$this->taxonomies = get_the_taxonomies($post);
+	}
+	//only run after set_taxonomies()
+	function set_categories ($post) {
+		if ($this->taxonomies == null) {
+			$this->set_taxonomies($post);
+		}  else {
+			$this->categories = strip_tags($this->taxonomies['category']);
+		}
+	}
+	function set_tags ($post) {
+		if ($this->taxonomies == null) {
+			$this->set_taxonomies($post);
+		} else {
+			$this->tags = strip_tags( $this->taxonomies['post_tag'] );
+		}
+	}
+	function set_guidance ($post) {
+		if ($this->taxonomies == null) {
+			$this->set_taxonomies($post);
+		} else {
+			$this->guidance = strip_tags($this->taxonomies['guidance']);
+		}
+	}
+	//can only run after set_guidance()
+	function set_recommended () {
+		if (!strpos($this->guidance, 'Recommended')) {
+			$this->recommended = 'false';
+		} else {
+			$this->recommended = 'true';
+		}
+	}
+	function set_online () {
+			if (!strpos($this->guidance, 'All online')) {
+			$this->online = 'false';
+		} else {
+			$this->online = 'true';
+		}
+	}
+	//can only run after set_categories()
+	function set_partners () {
+		if (strpos($this->categories, 'Findmypast.co.uk') !== false || strpos($this->categories, 'Ancestry.co.uk') !== false ) {
+			$this->partners = 'true';
+		} else {
+			$this->partners = 'false';
+		}
+	}
+
+	function get_url () {return $this->url;}
```

Comment:
If these aren't being used I'd either delete them or replace the `sprintf()` that's used to create `$compiled` with calls to the getters. Personally, I'd probably just delete them because they don't need to be exposed to other objects.